share/patch_postgresql_db_zonemaster_backend_ver_7.0.0.pl

```diff
@@ -20,8 +17,62 @@ sub patch_db {
     ####################################################################
     # TEST RESULTS
     ####################################################################
-    $dbh->do( 'ALTER TABLE test_results ADD COLUMN undelegated integer NOT NULL DEFAULT 0' );
+    eval {
+        $dbh->do( 'ALTER TABLE test_results ADD COLUMN undelegated integer NOT NULL DEFAULT 0' );
+    };
+    if ($@) {
+        print "Error while changing DB schema:  " . $@;
+    }
+
+    $dbh->do( qq[
```

Comment:
Could this be done using the much simpler for-loop from the MySQL and SQLite implementations?

Reply:
It could but it has a impact on the performance. Processing 13K rows (about 1% of the zonemaster.net database) with the sql statement takes around 1 second and doing it with the while loop takes 17 seconds.

Reply:
That's an impressive difference. But this is a one-time task.
And the SQL statement is quite complex. What are the chances that a bug has snuck into it?

Reply:
Fair enough, I can can change it to the other implementation.

Reply:
@blacksponge, have you addressed the comment from @mattias-p?

Reply:
It has now been addressed.